astropy/coordinates/tests/test_frames.py

```diff
@@ -1496,7 +1496,9 @@ class FrameWithCoordinateAttribute(BaseCoordinateFrame):
     # The input is already HCRS, so the frame attribute should not change it
     assert f1_frame.coord_attr == hcrs
     # The output should not be different if a SkyCoord is provided
-    assert f1_skycoord.coord_attr == f1_frame.coord_attr
+    assert isinstance(f1_skycoord.coord_attr, SkyCoord)
```

Comment:
This test may have to be adjusted given the above comment.

Reply:
Actually, the test can be left like it was before, so perfect!